src/java.desktop/windows/classes/sun/print/PrintServiceLookupProvider.java

```diff
@@ -116,6 +116,7 @@ private synchronized void refreshServices() {
         if (printers == null) {
```

Comment:
`getAllPrinterNames` returns null when an error occurs or when there are no printers in the system.

Up for discussion:
If an error occurs, we may leave the old services without invalidating them. In this case, it is necessary to distinguish between the error and the lack of printers in the system. Does this make any sense?